### PR TITLE
[Core] Enhance docs of options

### LINKED ISSUE #26767: [core] .options documentation is missing.
### Description

```
>>> import ray
>>> @ray.remote
... def f():
...     pass
...
>>> help(f.options)

Help on method options in module ray.remote_function:

options(**task_options) method of ray.remote_function.RemoteFunction instance
    Configures and overrides the task invocation parameters.

    The arguments are the same as those that can be passed to :obj:`ray.remote`.
    Overriding `max_calls` is not supported.

    Examples:

    .. code-block:: python

        @ray.remote(num_gpus=1, max_calls=1, num_returns=2)
        def f():
           return 1, 2
        # Task f will require 2 gpus instead of 1.
        g = f.options(num_gpus=2)
```

From the help, there is no way to tell what's the options there. 

### Link

_No response_

### PR DESCRIPTION
**Pull request by suquark**
Signed-off-by: Siyuan Zhuang <suquark@gmail.com>

## Why are these changes needed?

The doc of `.options()` is not detailed enough to show what options are allowed. See https://github.com/ray-project/ray/issues/26767

## Related issue number

Closes #26767

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/actor.py [hunk 1]
```diff
@@ -534,6 +534,73 @@ def options(self, **actor_options):
         The arguments are the same as those that can be passed
         to :obj:`ray.remote`.
 
+        Args:
```
Comment (iycheng) [inline python/ray/actor.py:537]: is it possible to add them and generate with option_utils? 
For example https://github.com/ray-project/ray/blob/master/python/ray/_private/ray_option_utils.py

we have task only options, actor only options, common options, I feel maybe the comment should be put there.

**Comment by iycheng** [comment]
I'm OK with this short term solution given that it's not a hi-pri things and mitigate the current use cases.

File: python/ray/actor.py (hunk 1 above)
Comment (iycheng) [inline python/ray/actor.py:537]: is it possible to add them and generate with option_utils? 
For example https://github.com/ray-project/ray/blob/master/python/ray/_private/ray_option_utils.py

we have task only options, actor only options, common options, I feel maybe the comment should be put there.

**Review by fishbone — APPROVED** [review]

File: python/ray/actor.py (hunk 1 above)
Comment (suquark) [reply python/ray/actor.py:537]: Yes, that could be a long-term solution, but it does have a cost: developers wouldn't get a direct doc by looking at `options`. Maybe we should generate the doc automatically with a script and checks if the doc is update-to-date with a test.

**Comment by suquark** [comment]
The windows CI failed, but I could not see how it is related to this PR. I'll merge it.

**Pull request merged**